### ISSUE TITLE
pep8 check breaks?

### ISSUE THREAD

**Issue by shwsun** [CCI-MOC/hil#650]
When I run the tests by hand, I got errors like below.

```sh
========================================== ERRORS ===========================================
_________________________________ ERROR collecting setup.py _________________________________
setup.py:98: in <module>
    'lxml>=3.6.0,<4.0',
/usr/lib64/python2.7/distutils/core.py:140: in setup
    raise SystemExit, gen_usage(dist.script_name) + "\nerror: %s" % msg
E   SystemExit: usage: -c [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
E      or: -c --help [cmd1 cmd2 ...]
E      or: -c --help-commands
E      or: -c cmd --help
E   
E   error: no commands supplied
________________________ ERROR collecting haas/ext/auth/keystone.py _________________________
haas/ext/auth/keystone.py:5: in <module>
    from keystonemiddleware.auth_token import filter_factory
E   ImportError: No module named keystonemiddleware.auth_token
__________________________ ERROR collecting haas/migrations/env.py __________________________
haas/migrations/env.py:18: in <module>
    config = context.config
E   AttributeError: 'module' object has no attribute 'config'
______________________ ERROR collecting tests/integration/keystone.py _______________________
tests/integration/keystone.py:17: in <module>
    from keystoneauth1.identity import v3
E   ImportError: No module named keystoneauth1.identity
____________________________ ERROR collecting tests/unit/cli.py _____________________________
import file mismatch:
imported module 'cli' has this __file__ attribute:
  /home/jethro/git/hil/tests/integration/cli.py
which is not the same as the test file we want to collect:
  /home/jethro/git/hil/tests/unit/cli.py
HINT: remove __pycache__ / .pyc files and/or use a unique basename for your test file modules
========================================= FAILURES ==========================================
________________________________________ PEP8-check _________________________________________
[gw1] linux2 -- Python 2.7.5 /home/jethro/git/hil/.venv/bin/python
/home/jethro/git/hil/haas/test_common.py:311:80: E501 line too long (85 > 79 characters)
            nic = db.session.query(Nic).filter_by(owner=node, label='boot-nic').one()
                                                                               ^

======================= 1 failed, 65 passed, 5 error in 7.65 seconds ========================
(.venv) [jethro@localhost: hil (master)]$ 
```

@zenhack This raises two concerns: 
  1. travis is not checking pep8 right now, at least the way we want
  2. should code reviewers doing these tests before giving a +1? I feel like I failed my job a little...

**Comment by zenhack** [comment]
Re: (2), We probably ought to be doing this *at least* if it's touching testing infrastructure; this used to happen more often before traivs, because people forget. But yeah, I feel like we screwed up.

I'm futzing with it locally trying to see what's necessary to fix this. Observations/issues:

* pytest depends on the ability to not just parse but *import* modules in order to scan for tests. This breaks with `haas/migrations/env.py`, because that script is supposed to be run in a specialized environment set up by alembic. This wasn't an issue before because we were only scanning tests/ for tests.
* pytest royally screws up namespacing (read: completely defeats every namespacing mechanism available in python and provides nothing to compensate), so it chokes on there being files with duplicate names *anywhere* in it's scan. Some of the errors are because e.g. we have a `cli.py` both in `tests/unit/` and in `haas/`. We could probably hack around this by invoking py.test multiple times as needed, but ug.
* For some reason the pep8 plugin seems to "erase" some flags that py.test normally understands, e.g. `-x`. wtf.
* pytest-pep8's own build is currently failing: https://drone.io/bitbucket.org/pytest-dev/pytest-pep8/latest Great job guys.

There seems to be a tool `pep8` that is independent of pytest (and I think a dependency of the `pytest-pep8` plugin anyway) that does the job perfectly. On my machine:

```
(.venv) [isd@rook haas]$ pep8 haas/ tests/ examples/ scripts/ *.py
haas/test_common.py:311:80: E501 line too long (85 > 79 characters)
(.venv) [isd@rook haas]$ echo $?
1
```

So my vote is we just add that, instead of using pytest-pep8.

**Comment by zenhack** [comment]
//cc @matsuuran

**Comment by shwsun** [comment]
Yes, I tested the functionality before using `pep8` and it works fine.

I also saw some failures or errors by simply running `py.test tests`. Is it because of this PR?

**Comment by zenhack** [comment]
@shwsun, can you paste the error and the command you're running? The only thing I see above is the fact that you don't have python-keystoneclient installed.

Worth noting some tests require special setup.

**Comment by shwsun** [comment]
`py.test --pep8 -m pep8 *.py haas/ tests/`

```sh
========================================== ERRORS ==========================================
________________________________ ERROR collecting setup.py _________________________________
setup.py:98: in <module>
    'lxml>=3.6.0,<4.0',
/usr/lib64/python2.7/distutils/core.py:139: in setup
    raise SystemExit, gen_usage(dist.script_name) + "\nerror: %s" % msg
E   SystemExit: usage: py.test [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
E      or: py.test --help [cmd1 cmd2 ...]
E      or: py.test --help-commands
E      or: py.test cmd --help
E   
E   error: option --pep8 not recognized
________________________ ERROR collecting haas/ext/auth/keystone.py ________________________
haas/ext/auth/keystone.py:5: in <module>
    from keystonemiddleware.auth_token import filter_factory
E   ImportError: No module named keystonemiddleware.auth_token
_________________________ ERROR collecting haas/migrations/env.py __________________________
haas/migrations/env.py:18: in <module>
    config = context.config
E   AttributeError: 'module' object has no attribute 'config'
____________________________ ERROR collecting tests/unit/cli.py ____________________________
import file mismatch:
imported module 'cli' has this __file__ attribute:
  /home/shwsun/coding/hil/tests/integration/cli.py
which is not the same as the test file we want to collect:
  /home/shwsun/coding/hil/tests/unit/cli.py
HINT: remove __pycache__ / .pyc files and/or use a unique basename for your test file modules
========================================= FAILURES =========================================
________________________________________ PEP8-check ________________________________________
/home/shwsun/coding/hil/haas/test_common.py:311:80: E501 line too long (85 > 79 characters)
            nic = db.session.query(Nic).filter_by(owner=node, label='boot-nic').one()
                                                                               ^

=========================== 386 tests deselected by "-m 'pep8'" ============================
=============== 1 failed, 65 passed, 386 deselected, 4 error in 7.99 seconds ===============

```

And the python-keystoneclient is installed.

**Comment by matsuuran** [comment]
@shwsun @zenhack I have a branch ready to go that takes out `pytest-pep8` and replaces it with `pep8`.  I still get an error when I run `py.test tests/` but it does not look related to anything pep8

**Comment by zenhack** [comment]
I would expect errors when running pytest on all of test/; many of the tests outside of tests/unit/ require special setup. And you should never pass Haas/ to pytest. So I think we're fine as far as the existing tests go.

**Comment by zenhack** [comment]
@matsuuran, great! Submit a PR.

**Comment by shwsun** [comment]
I see, but at least the `py.test --pep8 -m pep8` is not working as expected and we need to replace it with `pep8`. I also think we need to add python-keystoneclient in the requirements anyway.

**Comment by zenhack** [comment]
@shwsun, the keystone library is only required for the keystone author back end, and I don't want to make it a hard dependency for all installations. Travis installs the keystone stuff already.

**Comment by matsuuran** [comment]
fixed in #651
